src/modules/campath/exporter.rs

```diff
@@ -0,0 +1,83 @@
+use std::fs::OpenOptions;
+use std::io::{BufWriter, Write};
+
+use super::camio::{CamIO, ViewInfoCamIO};
+use super::common::ViewInfo;
+
+#[derive(Clone)]
+pub struct Exporter {
+    filename: String,
+    data: CamIO,
+}
+
+impl Exporter {
+    pub fn new(filename: String) -> Self {
+        Self {
+            filename,
+            data: CamIO {
+                campaths: Vec::new(),
+            },
+        }
+    }
+
+    fn header(&self) -> String {
+        "\
+advancedfx Cam
+version 2
+channels time xPosition yPosition zPositon xRotation yRotation zRotation fov
+DATA
+"
+        .to_string()
+    }
+
+    pub fn append_entry(
+        &mut self,
+        time: f64,
+        vieworg: [f64; 3],
+        viewangles: [f64; 3],
+        fov: f64,
+    ) -> &Self {
+        self.data.campaths.push(ViewInfoCamIO {
+            viewinfo: ViewInfo {
+                vieworg,
+                viewangles,
+            },
+            time,
+            fov,
+        });
+        self
+    }
+
+    fn entry_to_string(&self, idx: usize) -> String {
+        let curr = self.data.campaths[idx];
+        format!(
+            "{} {} {} {} {} {} {} {}\n",
+            curr.time,
+            curr.viewinfo.vieworg[0],
+            curr.viewinfo.vieworg[1],
+            curr.viewinfo.vieworg[2],
+            curr.viewinfo.viewangles[0],
+            curr.viewinfo.viewangles[1],
+            curr.viewinfo.viewangles[2],
+            curr.fov
+        )
+    }
+
+    pub fn write(&self) {
+        let file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .open(self.filename.clone())
+            .expect("Error: Cannot create .cam motion file.");
+        let mut file = BufWriter::new(file);
+
+        file.write_all(self.header().as_bytes())
+            .expect("Error: Cannot write motion into buffer.");
+        for idx in 0..self.data.campaths.len() {
+            file.write_all(self.entry_to_string(idx).as_bytes())
+                .expect("Error: Cannot write motion into buffer.");
+        }
+        file.flush()
```

Comment:
Why is this needed?

Reply:
Is that not how it works? I really have no idea how this bufwriter works so I guess it writes into a buffer (in memory) then I need to flush it to write to file. My intention is to write into memory before writing into file because in case of export 60fps or more (10-15fps is my usual to-go).

Reply:
I would expect it to flush on drop automatically. Can you check?

Reply:
Sorry, I should have checked the docs myself, too.

> It is critical to call [flush](https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.flush) before BufWriter<W> is dropped. Though dropping will attempt to [flush](https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.flush) the contents of the buffer, any errors that happen in the process of dropping will be ignored. Calling flush ensures that the buffer is empty and thus dropping will not even attempt file operations.

In that case it is indeed better to have an explicit flush() with expect(). Could you please add it back? Otherwise the PR looks good.